x-pack/plugins/notifications/server/services/email.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UnsecuredActionsClient } from '@kbn/actions-plugin/server/unsecured_actions_client/unsecured_actions_client';
+import type { IEmailService, PlainTextEmail } from './email_service';
+
+export class EmailService implements IEmailService {
+  constructor(
+    private requesterId: string,
+    private connectorId: string,
+    private actionsClient: UnsecuredActionsClient
+  ) {}
+
+  async sendPlainTextEmail(params: PlainTextEmail): Promise<void> {
+    const actions = params.to.map((to) => ({
+      id: this.connectorId,
+      spaceId: 'default', // TODO should be space agnostic?
+      apiKey: null, // not needed for Email connector
+      executionId: '???',
+      params: {
+        ...params,
+        to,
+      },
+    }));
+    return await this.actionsClient.bulkEnqueueExecution(this.requesterId, actions);
```

Comment:
I was thinking we will execute the email action immediately, [using the `execute()` method](https://github.com/elastic/kibana/blob/main/x-pack/plugins/actions/README.md?plain=1#L139), (I hope that does not create a saved object), instead of "enqueueing", which creates a saved object.

Do we gain something by enqueueing an action?

Reply:
Added a comment about that here: https://github.com/elastic/kibana/issues/140743#issuecomment-1280646070

Reply:
Following the discussion, it seems that the consensus is to queue the task and execute asynchronously from task manager.